pkg/userdata/rhel/provider.go

```diff
@@ -132,6 +132,7 @@ func (p Provider) UserData(req plugin.UserDataRequest) (string, error) {
 const userDataTemplate = `#cloud-config
 {{ if ne .CloudProviderName "aws" }}
 hostname: {{ .MachineSpec.Name }}
+fqdn: {{ .MachineSpec.Name }}
```

Comment:
This means it is included on all cloud providers by aws. If you want to set it on OS only I would suggest that you add a new case for it:
```go
{{ if .CloudProviderName "openstack" }}
fqdn: {{ .MachineSpec.Name }}
{{ end }}
```

Reply:
Have you actually check if the subscription has been removed after this change?

Reply:
Yes, works for OpenStack / OTC with a Satellite server.